src/physics_common.cpp

```diff
@@ -23,4 +24,160 @@ void russian_roulette(Particle* p)
   }
 }
 
+//==============================================================================
+// IMPORTANCE_SPLIT
+//==============================================================================
+
+void split_by_ratio(Particle* p, double importance_ratio) {
+  p->wgt_last_ = p->wgt_;
+  double importance_rounded = std::floor(importance_ratio);
+  
+  int num_splits = 0;
+  // how many splits - probabilistic 3/2 split 
+  //  half the time we get 3 and half we get 2
+  if ( prn() < importance_ratio - importance_rounded )
+    num_splits = int(importance_rounded);
+  else 
+    num_splits = int(std::ceil(importance_ratio));
+  // calculate the tru number of splits
+  p->wgt_ /= float(num_splits);
+  // note 1 -> n not 0->n
+  for ( int i = 1 ; i < num_splits ; i++ ) {
+    p->create_secondary(p->u(),p->E_,p->type_);
+  }
+
+  return;
+}
+
+//==============================================================================
+// ROULETTE
+//==============================================================================
+
+void roulette(Particle* p, double importance_ratio) {
```

Comment:
Recommend a variable name other than `importance_ratio`.  This is really a survival probability that may be determined in a variety of ways: `survival_prob`?